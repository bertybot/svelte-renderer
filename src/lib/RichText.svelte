<script lang="ts">
	import type { EmbedReferences, RichTextContent } from '@graphcms/rich-text-types';
	import type { NodeRendererType } from './types.js';

	import RenderElements from './RenderElements.svelte';

	type Props = {
		content: RichTextContent;
		references?: EmbedReferences | undefined;
	} & NodeRendererType

	let {
		content,
		references = undefined,
		...renderers
	}: Props = $props();

	//Derive renderers with defaults so user doesnt have to supply all of them
</script>

<RenderElements {content} {renderers} {references}>
</RenderElements>