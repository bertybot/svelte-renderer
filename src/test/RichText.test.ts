import type { RichTextContent } from '@graphcms/rich-text-types';

import RichText from '$lib/RichText.svelte';
import { describe, it, expect, afterEach } from 'vitest';
import { render } from '@testing-library/svelte';
import {
	defaultContent as content,
	imageContent,
	videoContent,
	listContent,
	iframeContent,
	inlineContent,
	emptyContent,
	embedAssetContent,
	simpleH1Content,
	tableContent
} from './content.js';
import TestRichText from './testComponents/TestRichText.svelte';
import TestRichTextClass from './testComponents/TestRichTextClass.svelte';


//Svelte does this spacing stuff that makes the snapshots look weird but these are the same tests
describe('rich-text-svelte-renderer', () => {
	afterEach(() => {
		document.body.innerHTML = '';
	});
	it('renders content', () => {
		render(RichText, {
			target: document.body,
			props: {
				content
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders content correctly if received a object with children', () => {
		const contentObject: RichTextContent = {
			children: [
				{
					type: 'paragraph',
					children: [
						{
							bold: true,
							text: 'Hello World!'
						}
					]
				}
			]
		};

		render(RichText, {
			target: document.body,
			props: {
				content: contentObject
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('should not render elements if received a object with empty children', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: emptyContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('should render a table', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: tableContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('should should render H1 with some text', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: simpleH1Content
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders content with custom elements', () => {
		render(RichText, {
			target: document.body,
			props: {
				content,
				renderers: {
					p: TestP,
					bold: TestBold
				}
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders inline content', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: inlineContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders inline content with custom renderers', () => {
		render(TestRichText, {
			target: document.body,
			props: {
				content: inlineContent,
			},
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders link', () => {
		const linkContent: RichTextContent = [
			{
				type: 'link',
				id: 'test',
				rel: 'noreferrer',
				href: 'https://hygraph.com',
				title: 'Hygraph website',
				className: 'text-white',
				openInNewTab: true,
				children: [
					{
						text: 'Hygraph'
					}
				]
			}
		];

		render(RichText, {
			target: document.body,
			props: {
				content: linkContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders iframe', () => {
		const iframeContent: RichTextContent = [
			{
				url: 'https://www.youtube.com/watch?v=Ylmd737tw5w',
				type: 'iframe',
				children: [
					{
						text: ''
					}
				]
			}
		];

		render(RichText, {
			target: document.body,
			props: {
				content: iframeContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders class', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: iframeContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders class with custom renderer', () => {
		render(TestRichTextClass, {
			target: document.body,
			props: {
				content: iframeContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders image', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: imageContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('removes the width and height attributes if they are set to 0', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: [
					{
						src: 'https://media.graphassets.com/bFyCrmvuQfO7n0l5ZmH5',
						type: 'image',
						title: 'logo.svg',
						width: 0,
						handle: 'mQeGmwkXTnqTfcgUXVr7',
						height: 0,
						children: [
							{
								text: ''
							}
						],
						mimeType: 'image/svg+xml'
					}
				]
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders image with custom renderer', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: imageContent,
				renderers: {
					img: TestImage
				}
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders video', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: videoContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('renders lists', () => {
		render(RichText, {
			target: document.body,
			props: {
				content: listContent
			}
		});
		expect(document.body).toMatchSnapshot();
	});

	it('should render HTML and JSX tags correctly', () => {
		const content: RichTextContent = [
			{ type: 'paragraph', children: [{ text: '<Test />', code: true }] }
		];

		render(RichText, {
			target: document.body,
			props: {
				content: content
			}
		});
		expect(document.body.innerHTML).toContain('<Test />');
	});

	it('should render empty text spaces', () => {
		const content: RichTextContent = [
			{
				type: 'paragraph',
				children: [
					{ text: 'Sweet black ' },
					{ bold: true, text: 'cap' },
					{ text: ' ' },
					{ text: 'with', underline: true },
					{ text: ' ' },
					{ text: 'embroidered', italic: true },
					{ text: ' ' },
					{ bold: true, text: 'Hygraph' },
					{ text: ' logo.' }
				]
			}
		];

		render(RichText, {
			target: document.body,
			props: {
				content: content
			}
		});

		expect(document.body).toMatchSnapshot();
	});

	it('should replace all \n in a string with <br /> elements', () => {
		const content: RichTextContent = [
			{
				type: 'paragraph',
				children: [
					{
						text: "Hello,\n⁠My name is joão pedro,\n⁠I'm testing a bug"
					}
				]
			}
		];

		render(RichText, {
			target: document.body,
			props: {
				content: content
			}
		});

		expect(document.body).toMatchSnapshot();
	});
});

describe('custom embeds and assets', () => {
	afterEach(() => {
		document.body.innerHTML = '';
	});
	it('should render video, image, and audio assets', () => {
		const references = [
			{
				id: 'cknjbzowggjo90b91kjisy03a',
				url: 'https://media.graphassets.com/dsQtt0ARqO28baaXbVy9',
				mimeType: 'image/png'
			},
			{
				id: 'ckrus0f14ao760b32mz2dwvgx',
				url: 'https://media.graphassets.com/7M0lXLdCQfeIDXnT2SVS',
				mimeType: 'video/mp4'
			},
			{
				id: 'ckryzom5si5vw0d78d13bnwix',
				url: 'https://media.graphassets.com/H9eZ7CISSBpAKxqdSwzg',
				mimeType: 'audio/mpeg'
			}
		];

		render(RichText, {
			target: document.body,
			props: {
				content: embedAssetContent,
				references
			}
		});

		expect(document.body).toMatchSnapshot();
	});
});
