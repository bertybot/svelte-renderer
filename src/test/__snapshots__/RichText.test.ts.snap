// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`custom embeds and assets > should render specific mimeType if favour of the mimeType group 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <img
    loading="lazy"
    title=""
  />
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div>
    custom video/mp4
  </div>
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <audio
    controls=""
    src="https://media.graphassets.com/H9eZ7CISSBpAKxqdSwzg"
    style="display: block; max-width: 100%; height: auto;"
  >
    <p>
      Your browser doesn't support HTML5 audio. Here is a 
      <a
        href="https://media.graphassets.com/H9eZ7CISSBpAKxqdSwzg"
      >
        link to the audio
      </a>
       instead.
    </p>
  </audio>
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  
  
</body>
`;

exports[`custom embeds and assets > should render video, image, and audio assets 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <img
    loading="lazy"
    title=""
  />
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <video
    controls=""
    height="100%"
    src=""
    title=""
    width="100%"
  >
    <p>
      Your browser doesn't support HTML5 video. Here is a  
      <a
        href=""
      >
        link to the video
      </a>
       instead.
    </p>
  </video>
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <audio
    controls=""
    src="https://media.graphassets.com/H9eZ7CISSBpAKxqdSwzg"
    style="display: block; max-width: 100%; height: auto;"
  >
    <p>
      Your browser doesn't support HTML5 audio. Here is a 
      <a
        href="https://media.graphassets.com/H9eZ7CISSBpAKxqdSwzg"
      >
        link to the audio
      </a>
       instead.
    </p>
  </audio>
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > removes the width and height attributes if they are set to 0 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <img
    loading="lazy"
    src="https://media.graphassets.com/bFyCrmvuQfO7n0l5ZmH5"
    title="logo.svg"
  />
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders class 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    class="test"
  >
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <p>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      wow
      
      
      
      
      
      
      
    </p>
    
    
    
    
    
    
    
    
    
    
    
    <!---->
  </div>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders class with custom renderer 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <section
    class="bg-white"
  >
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <p>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      wow
      
      
      
      
      
      
      
    </p>
    
    
    
    
    
    
    
    
    
    
    <!---->
  </section>
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders content 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <b>
      <!---->
      <!---->
      Hello World!
      
      
      
    </b>
    
    
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders content correctly if received a object with children 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <b>
      <!---->
      <!---->
      Hello World!
      
      
      
    </b>
    
    
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders content with custom elements 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p
    class="text-white"
  >
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <strong
      class="text-black"
    >
      <!---->
      Hello World!
      
      <!---->
    </strong>
    
    
    
    
    
    <!---->
  </p>
   ,
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders iframe 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <div
    style="position: relative; overflow: hidden; width: 100%; padding-top: 56.25%;"
  >
    <iframe
      allow="fullscreen"
      frameborder="0"
      loading="lazy"
      referrerpolicy="no-referrer"
      src="https://www.youtube.com/watch?v=Ylmd737tw5w"
      style="position: absolute; top: 0px; bottom: 0px; right: 0px; left: 0px; width: 100%; height: 100%;"
      title=""
    />
  </div>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders image 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <img
    alt="photo-1564631027894-5bdb17618445.jpg"
    height="1000"
    loading="lazy"
    src="https://media.graphassets.com/output=format:webp/resize=,width:667,height:1000/8xrjYm4CR721mAZ1YAoy"
    title="photo-1564631027894-5bdb17618445.jpg"
    width="667"
  />
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders image with custom renderer 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <img
    alt="photo-1564631027894-5bdb17618445.jpg"
    src="https://media.graphassets.com/output=format:webp/resize=,width:667,height:1000/8xrjYm4CR721mAZ1YAoy"
  />
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders inline content 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <b>
      <!---->
      <!---->
      Hey, 
      
      
      
    </b>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <i>
      <!---->
      <!---->
      how
      
      
      
    </i>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <u>
      <!---->
      <!---->
      are
      
      
      
    </u>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <code>
      <!---->
      <!---->
      you?
      
      
      
    </code>
    
    
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders inline content with custom renderers 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <strong
      class="text-black"
    >
      <!---->
      Hey, 
      
      <!---->
    </strong>
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <i
      class="italic-class"
      style="color: red;"
    >
      <!---->
      how
      
      <!---->
    </i>
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <u
      role="button"
    >
      <!---->
      are
      
      <!---->
       test
    </u>
    ,
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <code
      style="font-style: italic;"
    >
      <!---->
      you?
      
      <!---->
    </code>
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders link 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <a
    class="text-white"
    href="https://hygraph.com"
    id="test"
    rel="noreferrer"
    target="_blank"
    title="Hygraph website"
  >
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    Hygraph
    
    
    
    
    
    
    <!---->
  </a>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders lists 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <ul>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Embroided logo
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
    </li>
    
    
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Fits well
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
    </li>
    
    
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Comes in black
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
    </li>
    
    
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <li>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Reasonably priced
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      
    </li>
    
    
    
    
    
    
    
    
    
    
    
    
  </ul>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > renders video 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <video
    controls=""
    height="400"
    src="https://media.graphassets.com/oWd7OYr5Q5KGRJW9ujRO"
    title="file_example_MP4_480_1_5MG.m4v"
    width="400"
  >
    <p>
      Your browser doesn't support HTML5 video. Here is a  
      <a
        href="https://media.graphassets.com/oWd7OYr5Q5KGRJW9ujRO"
      >
        link to the video
      </a>
       instead.
    </p>
  </video>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should not render elements if received a object with empty children 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <h2>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <a
      href="https://hygraph.com"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Testing Link
      
      
      
      
      
      
      <!---->
    </a>
    
    
    
    
    
    
    
    
    
    
    
    
  </h2>
  
  
  
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <h2>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <a
      href="https://hygraph.com"
    >
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      Link
      
      
      
      
      
      
      <!---->
    </a>
    
    
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
     2
    
    
    
    
    
    
    
  </h2>
  
  
  
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  
  
  
  <!---->
  <!---->
  <!---->
  
  
  
  <!---->
  <!---->
  <!---->
  
  
  
  <!---->
  <!---->
  <!---->
  
  
  
  <!---->
  <!---->
  <!---->
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <table>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <tbody>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <tr>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <td>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 1 - Col 1
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </td>
        
        
        
        
        
        
        
        
        
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <td>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 1 - Col 2
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </td>
        
        
        
        
        
        
        
        
        
        
        
        
      </tr>
      
      
      
      
      
      
      
      
      
      
      
      
    </tbody>
    
    
    
    
    
    
    
    
    
    
    
    
  </table>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should render a table 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <table>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <thead>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <tr>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <th>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 1 - Header 1
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </th>
        
        
        
        
        
        
        
        
        
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <th>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 1 - Header 2
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </th>
        
        
        
        
        
        
        
        
        
        
        
        
      </tr>
      
      
      
      
      
      
      
      
      
      
      
      
    </thead>
    
    
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <tbody>
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <!---->
      <tr>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <td>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 2 - Col 1
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </td>
        
        
        
        
        
        
        
        
        
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <td>
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <!---->
          <p>
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            <!---->
            Row 2 - Col 2
            
            
            
            
            
            
            
          </p>
          
          
          
          
          
          
          
          
          
          
          
          
        </td>
        
        
        
        
        
        
        
        
        
        
        
        
      </tr>
      
      
      
      
      
      
      
      
      
      
      
      
    </tbody>
    
    
    
    
    
    
    
    
    
    
    
    
  </table>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should render empty text spaces 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    Sweet black 
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <b>
      <!---->
      <!---->
      cap
      
      
      
    </b>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
     
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <u>
      <!---->
      <!---->
      with
      
      
      
    </u>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
     
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <i>
      <!---->
      <!---->
      embroidered
      
      
      
    </i>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
     
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <b>
      <!---->
      <!---->
      Hygraph
      
      
      
    </b>
    
    
    
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
     logo.
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should render superscript and subscript elements 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    TEST
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <sub>
      <!---->
      <!---->
      T
      
      
      
    </sub>
    
    
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    TEST
    
    
    
    
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <sup>
      <!---->
      <!---->
      2
      
      
      
    </sup>
    
    
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should replace all 
 in a string with <br /> elements 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <p>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    Hello, 
    <br />
    <!---->
     My name is joão pedro, 
    <br />
    <!---->
     I'm testing a bug 
    <!---->
    
    
    
    
    
    
    
    
  </p>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;

exports[`rich-text-svelte-renderer > should should render H1 with some text 1`] = `
<body>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <h1>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    heading
    
    
    
    
    
    
    
  </h1>
  
  
  
  
  
  
  
  
  
  
  
</body>
`;
